client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx

```diff
@@ -95,7 +99,9 @@ const SiteMigrationInstructions: Step = function ( { navigation, flow } ) {
 		completed: preparationCompleted,
 		error: preparationError,
 		migrationKey,
-	} = usePrepareSiteForMigration( siteId );
+	} = config.isEnabled( 'migration-flow/enable-white-labeled-plugin' )
+		? usePrepareSiteForMigrationWithMigrateToWPCOM( siteId ) // eslint-disable-line react-hooks/rules-of-hooks -- Temporary workaround until we completely replace the migrate guru hook.
+		: usePrepareSiteForMigrationWithMigrateGuru( siteId ); // eslint-disable-line react-hooks/rules-of-hooks
```

Comment:
This is somewhat hacky but I don't think it can break the hook call order because the condition is a constant. Let me know if there's a better solution.

Note that this is expected to be cleaned up once the Migrate Guru hook is no longer used.

Reply:
I would suggest creating a ticket to improve it before launch.  
I already had an issue caused by it that was very hard to detect.  I can help with possible solutions

Reply:
Oh, thanks for pointing that out, Gabriel. I'm also sorry for any trouble it may have caused.

We're planning to clean this up in the following week or two. Should we prioritize fixing it before that? Is it causing any bugs right now?

Reply:
I've created a card for it: https://github.com/Automattic/wp-calypso/issues/94750.